src/java.base/share/classes/java/lang/String.java

```diff
@@ -689,12 +689,24 @@ private static String decode(Charset charset, byte[] bytes, int offset, int leng
     }
 
     /*
-     * Throws iae, instead of replacing, if malformed or unmappable.
-     * The byte array can be exclusively used to construct
-     * the string and is not modified or used for any other purpose.
+     * {@return a new string by decoding from the given UTF-8 bytes array}
+     * <p>
+     * <b>WARNING: The caller of this method is assumed to have relinquished
+     * and transferred the ownership of the byte array</b>. It can thus be
+     * exclusively used to construct the {@code String}.
+     *
+     * @param bytes byte array containing UTF-8 encoded characters
+     * @param offset the index of the first byte to decode
+     * @param length the number of bytes to decode
+     * @throws NullPointerException If {@code bytes} is null
+     * @throws StringIndexOutOfBoundsException If {@code offset} is negative,
+     *         {@code length} is negative, or {@code offset} is greater than
+     *         {@code bytes.length - length}
+     * @throws CharacterCodingException for malformed input or unmappable characters
      */
-    private static String newStringUTF8NoRepl(byte[] bytes, int offset, int length) {
-        checkBoundsOffCount(offset, length, bytes.length);
+    private static String newStringUTF8OrThrow(byte[] bytes, int offset, int length)
```

Comment:
Renaming to the OrThrow suffix is good, that really helps the use-sites, the far away ones in particular.